src/mds/Migrator.cc

```diff
@@ -1219,8 +1219,8 @@ void Migrator::check_export_size(CDir *dir, export_state_t& stat, set<client_t>&
     dfs.pop_front();
 
     approx_size += frag_size;
-    for (CDir::map_t::iterator p = dir->begin(); p != dir->end(); ++p) {
-      CDentry *dn = p->second;
+    for (auto &p : *dir) {
```

Comment:
this is confusing. how about changing it to "for (auto p = dir->begin(); p != dir->end(); ++p)"